core/src/main/java/org/elasticsearch/common/Booleans.java

```diff
@@ -56,77 +50,148 @@ public static boolean isBoolean(char[] text, int offset, int length) {
      */
-    public static boolean parseBoolean(String value, boolean defaultValue) {
+    @Deprecated
+    public static boolean parseBooleanLenient(String value, boolean defaultValue) {
         if (value == null) {
             return defaultValue;
         }
         return !(value.equals("false") || value.equals("0") || value.equals("off") || value.equals("no"));
     }
 
     /**
-     * Returns <code>true</code> iff the value is either of the following:
-     *   <tt>false</tt>, <tt>0</tt>, <tt>off</tt>, <tt>no</tt>
-     *   otherwise <code>false</code>
+     * @return <code>true</code> iff the value is <tt>false</tt>, otherwise <code>false</code>.
      */
-    public static boolean isExplicitFalse(String value) {
-        return value != null && (value.equals("false") || value.equals("0") || value.equals("off") || value.equals("no"));
+    public static boolean isFalse(String value) {
+        return "false".equals(value);
     }
 
     /**
-     * Returns <code>true</code> iff the value is either of the following:
-     *   <tt>true</tt>, <tt>1</tt>, <tt>on</tt>, <tt>yes</tt>
-     *   otherwise <code>false</code>
+     * @return <code>true</code> iff the value is <tt>true</tt>, otherwise <code>false</code>
      */
-    public static boolean isExplicitTrue(String value) {
-        return value != null && (value.equals("true") || value.equals("1") || value.equals("on") || value.equals("yes"));
+    public static boolean isTrue(String value) {
+        return "true".equals(value);
+    }
+
+    /**
+     * Returns <code>false</code> if text is in <tt>false</tt>, <tt>0</tt>, <tt>off</tt>, <tt>no</tt>; else, true
+     *
+     * @deprecated Only kept to provide automatic upgrades for pre 6.0 indices. Use {@link #parseBoolean(char[], int, int, boolean)} instead
+     */
+    @Deprecated
+    public static boolean parseBooleanLenient(char[] text, int offset, int length, boolean defaultValue) {
+        if (text == null || length == 0) {
+            return defaultValue;
+        }
+        if (length == 1) {
+            return text[offset] != '0';
+        }
+        if (length == 2) {
+            return !(text[offset] == 'n' && text[offset + 1] == 'o');
+        }
+        if (length == 3) {
+            return !(text[offset] == 'o' && text[offset + 1] == 'f' && text[offset + 2] == 'f');
+        }
+        if (length == 5) {
+            return !(text[offset] == 'f' && text[offset + 1] == 'a' && text[offset + 2] == 'l' && text[offset + 3] == 's' &&
+                text[offset + 4] == 'e');
+        }
+        return true;
+    }
+
+    /**
+     * returns true if the a sequence of chars is one of "true","false","on","off","yes","no","0","1"
+     *
+     * @param text   sequence to check
+     * @param offset offset to start
+     * @param length length to check
+     *
+     * @deprecated Only kept to provide automatic upgrades for pre 6.0 indices. Use {@link #isBoolean(char[], int, int)} instead.
```

Comment:
I checked usages of all `Booleans.*lenient()` but I didn't see the code you're referring to. However, if it's just about the wording of the Javadoc I'm fine with changing it. The gist of the Javadocs should be: "Don't use it, we just need it during a transition phase and it will be removed."